server.go

```diff
@@ -327,6 +334,31 @@ func (s *Server) handleClient(conn net.Conn) {
 		goto reject
 	}
 
+	// needs additional auth
+	if s.authHandler != nil {
+		token = resp.Header.Get("X-Auth-Header")
+		if token == "" {
+			err = errors.New("Auth header missing")
+			logger.Log(
+				"level", 2,
+				"msg", "handshake failed",
+				"err", err,
+			)
+			goto reject
+		}
+
+		authorized := s.authHandler(token)
```

Comment:
I don't know if it maybe makes more sense for authHandler to return error and then you can have all logic required in authHandler
Also you could proably check header in that function so you don't hardcode X-AUTH-HEADER if someoone wants to call it somehting else
Seems like more of a go aproach